src/main/java/org/example/userBook/UserBookToFile.java

```diff
@@ -0,0 +1,144 @@
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+@Component("userBookToFile")
+public class UserBookToFile implements UserBook {
+    private final String filePath;
+
+    public UserBookToFile(@Value("${file.path}") String filePath) {
+        this.filePath = filePath;
+    }
+
+    //    "UsersRecordsFile"
+    private void saveUserToFile(User user) throws IOException {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(filePath, true))) {
+            writer.write(user.toString());
+            writer.newLine();
+        }
+    }
+
+    private List<User> loadUsersFromFile() throws IOException {
+        List<User> users = new ArrayList<>();
+        try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                User user = parseUserFromString(line);
+                users.add(user);
+            }
+        }
+        return users;
+    }
+
+    private void deleteUserFromFile(Integer userId) throws IOException {
+        List<String> lines = new ArrayList<>();
+        try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                User user = parseUserFromString(line);
+                if (!user.getId().equals(userId)) {
+                    lines.add(line);
+                }
+            }
+        }
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(filePath))) {
+            for (String line : lines) {
+                writer.write(line);
+                writer.newLine();
+            }
+        }
+    }
+
+    public User getUserByIdFromFile(int userId) throws IOException {
+        try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                User user = parseUserFromString(line);
+                if (user.getId() == userId) {
+                    return user;
+                }
+            }
+        }
+        return null;
+    }
+
+    private User parseUserFromString(String line) {
+        String[] parts = line.split("\\|");
+        String name = parts[0];
+        String surName = parts[1];
+        int age = Integer.parseInt(parts[2]);
+        int id = Integer.parseInt(parts[3]);
+        return new User(name, surName, age, id);
+    }
+
+    private boolean doesUserExist(int userId, List<User> users) {
+        for (User existingUser : users) {
+            if (existingUser.getId().equals(userId)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public void addUser(User user) {
+        try {
+            List<User> users = loadUsersFromFile();
+            if (!doesUserExist(user.getId(), users)) {
```

Comment:
if you have 1M users and I try to add user with id = 1.  I will reads all users but actually read only first one would be enough.  Please, proposal any solutions to avoid unnecessary load from file to RAM.
Also, how two users with same id could be if your application generate ids ?